bugbug/repository.py

```diff
@@ -114,6 +117,10 @@ def download_commits(repo_dir):
     commits = hg_log(repo_dir)
     commits_num = len(commits)
 
+    global number_of_commits
+    for i in range(commits_num):
+        number_of_commits[commits[i]] = sum(commit.author == commits[i].author for commit in commits[:i])
```

Comment:
With this implementation, you're iterating through all commits multiple times.
It's better to iterate only once, having a temporary map where you store the number of commits by the author so far.